### DISCUSSION TITLE
pipenv install vs pipenv install --ignore-pipfile

### DISCUSSION THREAD

**Discussion by Victor-Villacis** [pypa/pipenv#5411] [General]
If I have a Pipfile and a Pipfile.lock when I use `pipenv install` or `pipenv install --ignore-pipfile` both commands install from the Pipfile.lock. Both ignore the Pipfile. Why? 

For example my Pipfile has a package **pandas = "*"** and my Pipfile.lock has  **"version": "==1.0.0"**  when I do am ready to set up a new environment and I do `pipenv install` I except for 1.5.0 to be installed because it is the latest. 

However pipenv installs from Pipfile.lock instead, which installs 1.0.0. If I wanted pipenv to install from the lock file I would have done `pipenv install --ignore-pipfile` granted I can do a `pipenv update` after, but my point is why have the ignore flag, if without it, its going to ignore it anyway.

**Comment by vishal180618** [comment]
you can use `pipenv install --ignore-pipfile` to install from `Pipfile` instead of `Pipfile.lock`.

**Comment by kalebmckale** [comment]
The use case, as I understand it, for `--ignore-pipfile` is when the `Pipfile` has changed, but you still want to install from previously locked `Pipfile.lock` instead of first re-creating `Pipfile.lock`. If you wish to install directly from `Pipfile` without having to change it and go through the lock process, you can use `pipenv install --skip-lock` instead.

**Reply by ricolxwz** [reply]
I think this is not true, `pipenv install --ignore-pipfile` will do `pipenv lock` + `pipenv sync`. So `Pipfile.lock` will always be first re-created when using this command, and then install packages from `Pipfile.lock`.
